src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpProxyCacheableRequestGroup.java

```diff
@@ -211,17 +226,15 @@ private void doRequest(
         this.groupRequest = new HttpCacheProxyGroupRequest(factory, this, request);
 
         groupRequest.doRequest(traceId);
+        request.cleanupRequestHeadersIfNecessary();
```

Comment:
It would be better to inform the `request` that the request has been sent, say via `onQueuedRequestSent`, then let the internal implementation of `request` decide to cleanup the headers, rather than calling that directly from here.

Reply:
Done.